model/task/task.go

```diff
@@ -444,12 +444,14 @@ func (t *Task) DependenciesMet(depCaches map[string]Task) (bool, error) {
 		return true, nil
 	}
 
+	if !t.DependenciesMetTime.IsZero() {
```

Comment:
nit: I'd argue for combining this with 443, but if it's too long that way then

Reply:
As written, I agree, but I had to add another condition to capture restarted tasks in addition to tasks whose DependenciesMetTime hasn't been set.